### ISSUE TITLE
Jell-o text rendering (please see screenshots)

### ISSUE THREAD

**Issue by Slos** [mathjax/MathJax#2866]
```
<script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>

 <p>Rate of descent (RoD) is a function of descent angle and velocity.</p>

    <p>$$\text{Rate of Descent} = \frac{\left (D \times V\right) - \left (T \times V\right )}{W}$$</p>

    <p>therefore:</p>

    <p style="margin-top: 40px;;">$$\text{Rate of Descent} = \frac{\text{Power Required} - \text{Power Available}}{W}$$</p>
```

<img width="712" alt="image" src="https://user-images.githubusercontent.com/9729859/166306142-9d5e1af8-a119-42cf-9006-760ea326427a.png">

<img width="280" alt="image" src="https://user-images.githubusercontent.com/9729859/166306047-788ed163-f656-4774-be5c-97773e55e452.png">

<img width="575" alt="image" src="https://user-images.githubusercontent.com/9729859/166306094-bee0a93f-6143-4193-8516-26fb27269099.png">

**Comment by dpvc** [comment]
Yes, Safari has always had some difficulty with round-off errors, even though all the characters have the same baseline.

One work-around would be to use a system font for the contents of `\text{}` rather than the MathJax math fonts.  You can either inherit the surrounding text font or force a specific one.  E.g.,

``` js
MathJax = {
  chtml: {
    mtextInheritFont: true
  }
}
```

or

``` js
MathJax = {
  chtml: {
    mtextFont: Times
  }
}
```

See the [documentation](https://docs.mathjax.org/en/latest/options/output/index.html#options-common-to-all-output-processors) for more.

**Comment by benchristel** [comment]
I found that using a system font causes incorrect rendering when the text is embedded in a more complex expression.

For example, `\dfrac{2~\text{meters}}{3~\text{seconds}}` renders like this:

<img width="127" alt="Screen Shot 2023-03-07 at 11 33 50 AM" src="https://user-images.githubusercontent.com/693920/223533102-6f6f6d3d-2009-42c4-ae5d-8c6cc4230642.png">

Accents from the `textmacros` package cause the output to be illegible. E.g. `\text{G\"odel}` renders as:

<img width="62" alt="Screen Shot 2023-03-07 at 11 35 58 AM" src="https://user-images.githubusercontent.com/693920/223533555-57f2e38e-b421-4e31-9b0f-7d9ce5e4f55a.png">

I found another possible workaround for this issue, though: hardcoding the height of text characters.

```css
mjx-mtext > mjx-c::before {
    padding-top: 0.75em !important;
}
```

Text rendered in Safari 15.6.1 on macOS 12.6 with the CSS workaround above:

<img width="241" alt="Screen Shot 2023-03-07 at 11 25 37 AM" src="https://user-images.githubusercontent.com/693920/223530425-af609297-18d6-4698-81e5-fc433fe7d31d.png">

Without the workaround:

<img width="224" alt="Screen Shot 2023-03-07 at 11 26 23 AM" src="https://user-images.githubusercontent.com/693920/223530690-9e4a4c4c-4ec0-4cda-b478-f98ad85478f4.png">

I'm very surprised that this works. It causes accents from the `textmacros` package to display in the wrong place, though for me that's a minor issue, and can be hacked around with more CSS. What other issues might this workaround cause?

**Comment by dpvc** [comment]
I am not able to reproduce your first two examples.  Are you pre-rendering the pages on the server, or is MathJax running in the browser?  If in the browser, is the content initially in a container with `display:none`?  It looks like MathJax is not able to measure the width of the text, which is usually due to one of those situations.

The `padding-top` solution is one I've suggested as a last-ditch option for several people (which may be where you got it), and that certainly does get the text to align.  But it also has other ramifications, as you found with the accents, and it can cause vertical alignment issues for other situations, like `\sqrt{\mtext{a}}` for example, or `\text{a}^2`.  But if you have limited use of `\text`, and it is mostly for explanations and such, then it should be OK.

**Comment by benchristel** [comment]
Thanks! I am rendering on the server, so that is probably the explanation for what I was seeing.

**Comment by dpvc** [comment]
> I am rendering on the server, so that is probably the explanation for what I was seeing.

Yes, that would account for it, as fonts other than the ones provided by MathJax can't be measured there.  You could consider using a puppeteer-based solution, however, since that uses a headless Chrome that does handle the document layout just as Chrome does, and so the measurements will be possible in that case.  There is an [example](https://github.com/mathjax/MathJax-demos-node/tree/master/puppeteer) in the [MathJax node demos](https://github.com/mathjax/MathJax-demos-node) repository that might serve as a starting point if you want to try that out.
